### ISSUE TITLE
Localized bundle contains other languages files as well as per `webpack-bundle-analyzer` report

### ISSUE THREAD

**Issue by ahmadalfy** [angular/angular-cli#18839]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

I don't know

### Description

I tried to run webpack bundle analyzer to see why my main bundle is exceeding 2MB. I found that the main bundle contains the two locales I use in my application:

![image](https://user-images.githubusercontent.com/87611/93838861-6c8c2900-fc8b-11ea-819f-23fb92fc94dc.png)

As you can see, this bundle is for the English localized version and it contain `messages.ar.xlf`.

## 🔬 Minimal Reproduction

running the following command:

```
ng build --prod --localize --stats-json
```

successfully generates 2 different bundles as you can see in the following screenshot

![image](https://user-images.githubusercontent.com/87611/93838955-a65d2f80-fc8b-11ea-8ee7-6a643dc9e91f.png)

Yet, when I analyze the English bundle I see the Arabic is a part of the bundle itself

## 🌍 Your Environment

```
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 10.1.2
Node: 10.20.1
OS: darwin x64

Angular: 10.1.2
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, platform-server, router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1001.2
@angular-devkit/build-angular      0.1001.2
@angular-devkit/core               10.1.2
@angular-devkit/schematics         10.1.2
@angular/cdk                       10.2.2
@angular/material                  10.2.2
@angular/material-moment-adapter   10.2.2
@nguniversal/builders              10.1.0
@nguniversal/express-engine        10.1.0
@schematics/angular                10.1.2
@schematics/update                 0.1001.2
rxjs                               6.6.3
typescript                         3.9.7
```

**Comment by ahmadalfy** [comment]
Relevant part of `app.module.ts`

```javascript
import {
  NgModule,
  TRANSLATIONS,
  TRANSLATIONS_FORMAT,
  LOCALE_ID,
} from '@angular/core';
import { I18n } from '@ngx-translate/i18n-polyfill';

@NgModule({
  providers: [
    I18n,
    {
      provide: TRANSLATIONS,
      useFactory: (locale: string) => {
        locale = locale.split('-')[0] || 'en'; // default to english if no locale provided
        return require(`raw-loader!../locale/messages.${locale}.xlf`).default;
      },
      deps: [LOCALE_ID],
    },
    { provide: TRANSLATIONS_FORMAT, useValue: 'xlf' },
  ],
})
export class AppModule {}
```

**Comment by alan-agius4** [comment]
Hi @ahmadalfy,

When using Angular localize `--localize`, you don't need to `require` the XLF file in your application using Webpack `raw-loader`. 

With the above Webpack logic it is expected that both xlf files are included in the final bundle. 

NB: Using Webpack specific syntax such as the above is not supported by the Angular tooling team.

**Comment by ahmadalfy** [comment]
@alan-agius4 I want to thank you because your feedback started a journey of discovery. I found that I was using a package called [@ngx-translate/i18n-polyfill](https://github.com/ngx-translate/i18n-polyfill) which became obsolete starting from version 9.

I successfully removed it, migrated my code to use the new method `$localize` and properly configured the comparison of the multiple xliff files using [@ngx-i18nsupport/tooling](https://github.com/martinroob/ngx-i18nsupport/tree/master/projects/tooling) to generate different translation bundles.

Surprisingly the main bundle size is reduced by 1MB! from 2.3MB to 1MB. The webpack bundle analyzer report is great as well:

![image](https://user-images.githubusercontent.com/87611/93964406-c1996f00-fd5f-11ea-9e96-931e3ed47bd8.png)

Took quite some time and a lot of edits (160+ files) but it worked finally!

Thank you again
